### PR TITLE
Added Accumulo and Hadoop compat checks

### PR DESCRIPTION
**Pull request by keith-turner**

### CODE REVIEW

File: conf/uno.conf [hunk 1]
```diff
@@ -91,6 +91,27 @@ if [[ -n "$FLUO_YARN_REPO" ]]; then
...
+# ----------------------------
+# Apache built Accumuo 1.x only works with Hadoop 2.x.  Apache built Accumulo
+# 2.x only works with Hadoop 3.x.  However if building Accumulo, these
+# constraints may not apply.  The following sanity checks fail fast when
+# versions are incompatible.  If the checks cause a problem, comment them out
+# and consider submitting an issue or PR.
+#
+
+if [[ -z "$ACCUMULO_REPO" ]]; then
+  if [[ "${ACCUMULO_VERSION}" = 1.* && ! "${HADOOP_VERSION}" = 2.* ]]; then
```
Comment (busbey) [inline conf/uno.conf]: I presume we're not trying to support versions of Accumulo old enough for it to matter that this is only true for 1.6+?

File: conf/uno.conf [hunk 2]
```diff
@@ -91,6 +91,27 @@ if [[ -n "$FLUO_YARN_REPO" ]]; then
...
+    echo "ERROR from uno.conf : When using Accumuo 1.x, expect Hadoop 2.x not $HADOOP_VERSION"
```
Comment (mikewalch) [inline conf/uno.conf]: should be `Accumulo`

File: conf/uno.conf [hunk 3]
```diff
@@ -91,6 +91,27 @@ if [[ -n "$FLUO_YARN_REPO" ]]; then
...
+    exit 1
+  fi
+
+  if [[ "${ACCUMULO_VERSION}" = 2.* && ! "${HADOOP_VERSION}" = 3.* ]]; then
+    echo "ERROR from uno.conf : When using Accumuo 2.x, expect Hadoop 3.x not $HADOOP_VERSION"
```
Comment (mikewalch) [inline conf/uno.conf]: Should be `Accumulo`

File: conf/uno.conf (hunk 1 above)
Comment (keith-turner) [reply conf/uno.conf]: I have no interest in supporting older than 1.7 at this point.  I added a check for hadoop 1.x.

**Comment by keith-turner** [comment]
@mikewalch I made a change to the checks after you approved.  I realized the checks for accumulo 2 were true even when building, so I generalized that.

**Comment by mikewalch** [comment]
updates look good to me

**Pull request merged**